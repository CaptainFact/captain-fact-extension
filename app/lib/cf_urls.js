import { CF_FRONT_URL } from './constants'

/**
 * Generate an URL for the "Add Video" form pre-filled with a provider URL
 * @param {string} providerURL - The YouTube URL
 */
export const linkToAddVideo = (providerURL) => {
  return `${CF_FRONT_URL}/videos/add?url=${encodeURIComponent(providerURL)}`
}

/**
 * Generate an URL for the fact-checking page for given video hash id
 * @param {string} hashID - Video hash ID
 */
export const linkToVerificationsPage = (hashID) => {
  return `${CF_FRONT_URL}/videos/${hashID}`
}
