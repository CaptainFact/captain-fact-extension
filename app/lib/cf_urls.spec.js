import { linkToAddVideo, linkToVerificationsPage } from './cf_urls'


test('linkToAddVideo', () => {
  snapshot(linkToAddVideo('https://www.youtube.com/watch?v=dQw4w9WgXcQ'))
})

test('linkToVerificationsPage', () => {
  snapshot(linkToVerificationsPage('xxxx'))
})
