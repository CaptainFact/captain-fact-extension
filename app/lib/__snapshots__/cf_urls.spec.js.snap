// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linkToAddVideo 1`] = `"http://TEST_FRONTEND/videos/add?url=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DdQw4w9WgXcQ"`;

exports[`linkToVerificationsPage 1`] = `"http://TEST_FRONTEND/videos/xxxx"`;
